spark-dependencies/pom.xml

```diff
@@ -776,6 +818,14 @@
     </profile>
 
     <profile>
+      <!--only used for pyspark/sparkR needed spark dist is downloaded, no need to download auto-->
+      <id>spark-dist-cached</id>
```

Comment:
Introducing such profile totally makes sense to enable opt-out behavior when such things as spark are not needed.

But I think we need to make sure it is `activeByDefault ` and point to `validate` phase, to have a build behavior compatible with current one, as our CI infrastructure relies on it for tests.